index.js

```diff
@@ -158,39 +159,48 @@ function _make_generateLog(metrics, func, start_time, config, context) {
           log('IOPIPE-DEBUG: ', JSON.stringify(response_body))
         }
 
-        var req = https.request({
-          hostname: config.url,
-          path: config.path,
-          port: 443,
-          method: 'POST',
-          headers: {'content-type' : 'application/json'},
-          agent: httpsAgent,
-          timeout: config.networkTimeout
-        }, (res) => {
-          var apiResponse = ''
-
-          res.on('data', function (chunk) {
-            apiResponse += chunk
-          })
+        dnsPromise.then((ipAddress) => {
+          var req = https.request({
+            hostname: ipAddress,
```

Comment:
What will happen across multiple requests? Isn't the promise in the resolved state once? And what happens if the DNS changes or that IP goes down or if we want to change what IP the DNS is pointing to?